sentry/src/main/java/io/sentry/EventProcessor.java

```diff
@@ -3,7 +3,12 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+// other approach is to use EventProcessor<T extends SentryBaseEvent>
+// and having a processor which is the common fields and specific processors for Event and Transactions
+// not sure if our event processor interface should be deal with this
+// the downside using only the base class, if people want to process a field which is not part of SentryBaseEvent
+// they need to cast it manually
```

Comment:
@bruno-garcia can we discuss this

Reply:
after the discussion, we could try to use default methods for interfaces, return it so it's not a breaking change.